File: fanda/src/services/remote/foodService.js
import { get, post } from '@/utils/http'
export default {
  async getAll() { return get('/api/foods') },
  async getByCategory(category) { return get('/api/foods', { category }) },
  async search(keyword) { return get('/api/foods', { keyword }) },
  async addCustom(food) { return post('/api/foods', food) },
}

File: fanda-server/src/main/java/com/fanda/service/FoodService.java
package com.fanda.service;

import com.fanda.dto.request.FoodCreateRequest;
import com.fanda.dto.response.FoodItemResponse;

import java.util.List;

public interface FoodService {
    List<FoodItemResponse> getAll(Long userId);
    List<FoodItemResponse> getByCategory(String category, Long userId);
    List<FoodItemResponse> search(String keyword, Long userId);
    FoodItemResponse addCustom(FoodCreateRequest request, Long userId);
}
